File: middleware/fastapi/test_middleware.py
"""
Simple test to verify FastAPI middleware works with the updated SDK
"""

import asyncio
from fastapi import FastAPI, Request
from agent_passport_middleware import (
    AgentPassportMiddleware,
    AgentPassportMiddlewareOptions,
    require_policy,
    require_refund_policy
)

app = FastAPI()

# Test global middleware
app.add_middleware(
    AgentPassportMiddleware,
    options=AgentPassportMiddlewareOptions(
        base_url="https://api.aport.io",
        fail_closed=False,  # Don't fail for testing
        skip_paths=["/health", "/test"]
    )
)

# Test route-specific middleware
@app.post("/api/refunds")
async def process_refund(request: Request):
    # Add policy middleware
    middleware = require_refund_policy("test-agent-id")
    response = await middleware(request, lambda req: None)
    if response:
        return response
    
    return {
        "success": True,
        "message": "Refund processed",
        "agent_id": getattr(request.state, 'agent', {}).get('agent_id'),
        "policy_result": getattr(request.state, 'policy_result', None)
    }

# Health check
@app.get("/health")
async def health_check():
    return {"status": "ok"}

# Test endpoint
@app.get("/test")
async def test_endpoint():
    return {"message": "Middleware test endpoint"}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)

File: middleware/express/test-middleware.js
/**
 * Simple test to verify Express middleware works with the updated SDK
 */

const express = require("express");
const { agentPassportMiddleware, requirePolicy } = require("./dist/index.js");

const app = express();
app.use(express.json());

// Test global middleware
app.use(
  agentPassportMiddleware({
    baseUrl: "https://api.aport.io",
    failClosed: false, // Don't fail for testing
    skipPaths: ["/health", "/test"],
  })
);

// Test route-specific middleware
app.post(
  "/api/refunds",
  requirePolicy("finance.payment.refund.v1", "test-agent-id"),
  (req, res) => {
    res.json({
      success: true,
      message: "Refund processed",
      agent_id: req.agent?.agent_id,
      policy_result: req.policyResult,
    });
  }
);

// Health check
app.get("/health", (req, res) => {
  res.json({ status: "ok" });
});

// Test endpoint
app.get("/test", (req, res) => {
  res.json({ message: "Middleware test endpoint" });
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Express middleware test server running on port ${PORT}`);
  console.log("Test endpoints:");
  console.log(`  GET  http://localhost:${PORT}/health`);
  console.log(`  GET  http://localhost:${PORT}/test`);
  console.log(`  POST http://localhost:${PORT}/api/refunds`);
});
